### ISSUE TITLE
Repast models not working under macOS Catalina

### ISSUE THREAD

**Issue by rmellema** [Repast/repast.simphony#27]
I recently installed repast simphony on my machine running macOS Catalina, and everything works fine until I want to run a model. I have tried this with a few of them, but let's take heatbugs as an example. I can import the project just fine, and I can use the run button in Eclipse to build it, and have the Repast Simphony window show up. But then when I click the "start run" button, the application crashes "due to an uncaught exception 'NSInternalInconsistencyException'" (full log included below). 

From your FAQ I can see that this error is not unique to my setup, since it is mentioned there as a warning that can safely be ignored on Mojave. My guess is that this warning became more severe in Catalina, such that it now blocks the running of a model.

Java version is openjdk 13.0.2, OS is macOS Catalina 10.15.4, Repast is version 2.7.0.

```
2020-06-12 16:16:35.932 java[44797:16293652] CoreText note: Client requested name ".SFNS-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.932 java[44797:16293652] CoreText note: Set a breakpoint on CTFontLogSystemFontNameRequest to debug.
2020-06-12 16:16:35.933 java[44797:16293652] CoreText note: Client requested name ".SFNS-Bold", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.933 java[44797:16293652] CoreText note: Client requested name ".SFNSMono-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.937 java[44797:16293652] CoreText note: Client requested name ".SFNSMono-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.938 java[44797:16293652] CoreText note: Client requested name ".SFNSMono-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.938 java[44797:16293652] CoreText note: Client requested name ".SFNS-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2020-06-12 16:16:35.938 java[44797:16293652] CoreText note: Client requested name ".SFNS-Bold", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.core.util.Fields (file:/private/var/folders/k8/3h600g1s5hnbc8lkqvqpqv540000gp/T/AppTranslocation/797D8BFA-18BC-464A-A293-09315D86FCBF/d/Eclipse.app/Contents/Eclipse/plugins/repast.simphony.runtime_2.7.0/lib/xstream-1.4.7.jar) to field java.util.TreeMap.comparator
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.core.util.Fields
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2020-06-12 16:17:45.473 java[44797:16293652] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff368b5d07 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6f5d95bf objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff368de50c -[NSException raise] + 9
	3   AppKit                              0x00007fff33ae524c -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
	4   AppKit                              0x00007fff33acccb2 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
	5   AppKit                              0x00007fff33acc723 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	6   libnativewindow_macosx.jnilib       0x0000000131621f9e Java_jogamp_nativewindow_macosx_OSXUtil_CreateNSWindow0 + 398
	7   ???                                 0x00000001105d8c90 0x0 + 4569533584
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by ncollier** [comment]
Can you try with jdk 11? I've confirmed the models work as expected with those jdks under MacOS 10.15.4.

**Comment by rmellema** [comment]
I've installed the adopt open jdk 11 (version 11.0.7,10.1), set that as the default in Eclipse, set the compiler compliance level to version 11, and made sure that the run configuration used that jdk, but I get the same error.
The warnings are slightly different (no mention of fonts), so I've included the new stacktrace:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.core.util.Fields (file:/private/var/folders/k8/3h600g1s5hnbc8lkqvqpqv540000gp/T/AppTranslocation/E23C4522-0921-4477-92B4-8D2C04CBB331/d/Eclipse.app/Contents/Eclipse/plugins/repast.simphony.runtime_2.7.0/lib/xstream-1.4.7.jar) to field java.util.TreeMap.comparator
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.core.util.Fields
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2020-07-03 09:28:13.896 java[5261:53586104] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff343adbe7 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6d1855bf objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff343d63dc -[NSException raise] + 9
	3   AppKit                              0x00007fff315d4ddc -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
	4   AppKit                              0x00007fff315bc842 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
	5   AppKit                              0x00007fff315bc2b3 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	6   libnativewindow_macosx.jnilib       0x0000000169c5ff9e Java_jogamp_nativewindow_macosx_OSXUtil_CreateNSWindow0 + 398
	7   ???                                 0x000000011c5c0950 0x0 + 4770761040
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by etatara** [comment]
There is an incompatibility with the 3D displays and JDK 11+ in macos. If the 3D displays are needed under macos, then we recommend using JDK 8.  Other users have reported that JDK 9/10 also work in case Java 9+ language standards are needed.  If JDK 11+ is needed, then the only option is to disabled the 3D displays in your model.  All of the other Repast functionality aside from the 3D display works with JDK 11+.

**Comment by rmellema** [comment]
I don't know if this has anything to do with 3d, since these are the default provided models in repast (so not my own), but I decided to run under (adoptopen) JDK 8 (1.8.0_252) as well, with the same result:

```
2020-07-06 10:48:56.181 java[18984:58882044] Apple AWT Internal Exception: NSWindow drag regions should only be invalidated on the Main Thread!
2020-07-06 10:48:56.181 java[18984:58882044] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff343adbe7 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6d1855bf objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff343d63dc -[NSException raise] + 9
	3   AppKit                              0x00007fff315d4ddc -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
	4   AppKit                              0x00007fff315bc842 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
	5   AppKit                              0x00007fff315bc2b3 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	6   libnativewindow_macosx.jnilib       0x000000015fc69f9e Java_jogamp_nativewindow_macosx_OSXUtil_CreateNSWindow0 + 398
	7   ???                                 0x000000011699a407 0x0 + 4674135047
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by ncollier** [comment]
Can you try with an earlier version of 11 or 8? I've tested successfully with openjdk 1.8.0_232 and 11+28. I'm not sure at what point the warning became an exception.  In the meantime, we will try an updated version of the jogl library (which is doing the actual window creation).

**Comment by rmellema** [comment]
I managed to install openjdk 1.8.0_232, and it does work in that version. I cannot find adoptopenjdk changelogs, so I do not know where this change happened.
I hope an updated version of the jogl library helps, because it would be good not to have to rely on older java versions. :)

**Comment by ncollier** [comment]
Fixed with jogl-2.4 in a1322ea. We should close this after some more testing on windows.

**Comment by ncollier** [comment]
Merged into development bf68b431
